modules/dynamic-plugins/rhdh-supported-plugins.sh

```diff
@@ -301,52 +337,41 @@ if [[ $QUIET -eq 0 ]]; then
 fi
 
 # create .csv file with header
-echo -e "\"Name\",\"Plugin\",\"Role\",\"Version\",\"Support Level\",\"Path\",\"Required Variables\",\"Default\"" > "${0/.sh/.csv}"
+echo -e "\"Name\",\"Plugin\",\"Role\",\"Version\",\"Support Level\",\"Lifecycle\",\"Path\",\"Required Variables\",\"Default\"" > "${0/.sh/.csv}"
 
 num_plugins=()
-# append to .csv and .adocN files
-rm -f "${0/.sh/.adoc1}"
-sorted=(); while IFS= read -rd '' key; do sorted+=( "$key" ); done < <(printf '%s\0' "${!adoc1[@]}" | sort -z)
-if [[ $sorted ]] ;then
-  for key in "${sorted[@]}"; do
-      (( c = c + 1 ))
-      if [[ $QUIET -eq 0 ]]; then echo " * [$c] $key [ supported-plugins ] = ${csv[$key]}"; fi
-      echo -e "${adoc1[$key]}" >> "${0/.sh/.ref-rh-supported-plugins}"
-      # RHIDP-4196 omit techdocs plugins from the .csv
-      if [[ $key != *"techdocs"* ]]; then
-        echo -e "${csv[$key]}" >>  "${0/.sh/.csv}"
-      else
-        if [[ $QUIET -eq 0 ]]; then echo -e "${blue}   [WARN] Omit plugin $key from .csv file${norm}"; fi
-      fi
-  done
-fi
-num_plugins+=(${#adoc1[@]})
-
-rm -f "${0/.sh/.adoc2}"
-sorted=(); while IFS= read -rd '' key; do sorted+=( "$key" ); done < <(printf '%s\0' "${!adoc2[@]}" | sort -z)
-# shellcheck disable=SC2128
-if [[ $sorted ]] ;then
-    for key in "${sorted[@]}"; do
-      (( c = c + 1 ))
-      if [[ $QUIET -eq 0 ]]; then echo " * [$c] $key [ tech-preview-plugins ] = ${csv[$key]}"; fi
-      echo -e "${adoc2[$key]}" >> "${0/.sh/.ref-rh-tech-preview-plugins}";
-      echo -e "${csv[$key]}" >>  "${0/.sh/.csv}"
-    done
-fi
-num_plugins+=(${#adoc2[@]})
-
-rm -f "${0/.sh/.adoc3}"
-sorted=(); while IFS= read -rd '' key; do sorted+=( "$key" ); done < <(printf '%s\0' "${!adoc3[@]}" | sort -z)
-# shellcheck disable=SC2128
-if [[ $sorted ]] ;then
-    for key in "${sorted[@]}"; do
-      (( c = c + 1 ))
-      if [[ $QUIET -eq 0 ]]; then echo " * [$c] $key [ community-plugins ] = ${csv[$key]}"; fi
-      echo -e "${adoc3[$key]}" >> "${0/.sh/.ref-community-plugins}";
-      echo -e "${csv[$key]}" >>  "${0/.sh/.csv}"
+# Process temporary files instead of associative arrays
+for i in 1 2 3; do
```

Comment:
if we're switching to files why not use target file names instead of `adoc1.tmp` and this `1 2 3` loop?